lightningd/channel/channel.c

```diff
@@ -313,14 +313,14 @@ static void handle_peer_add_htlc(struct peer *peer, const u8 *msg)
 {
 	struct channel_id channel_id;
 	u64 id;
-	u32 amount_msat;
+	u64 amount_msat;
 	u32 cltv_expiry;
 	struct sha256 payment_hash;
 	u8 onion_routing_packet[TOTAL_PACKET_SIZE];
 	enum channel_add_err add_err;
 
 	if (!fromwire_update_add_htlc(msg, NULL, &channel_id, &id, &amount_msat,
-				      &cltv_expiry, &payment_hash,
+				      &payment_hash, &cltv_expiry,
```

Comment:
Lucky we didn't reorder two fields that have the same type, that'd be a nasty bug to trace down.